pipeline/transform/matches.py

```diff
@@ -24,7 +30,10 @@ def convert_matches_from_df(df: pd.DataFrame) -> List[Match]:
     cols = set(df.columns)
     field_names = [f.name for f in dataclasses.fields(Match) if f.name in cols]
     user_dicts = df[field_names].to_dict(orient="records")
-    return [Match(**u) for u in user_dicts]
+    matches = [Match(**u) for u in user_dicts]
```

Comment:
It would be better to parse the metadata before the `Match` objects are created, rather than editing it after the fact. You could change this list comprehension to a loop:

```python
matches = []
for record in match_dicts:
  # Some matches don't have metadata, so we should get an empty dictionary in that case
  raw_metadata = record.get("metadata", {})
  # There may be additional parsing needed for some of the more complicated types, but this should work for now, since you only depend on the generator field
  parsed_metadata = MatchMetadata(**raw_metadata)
  # We can override the metadata field with our new parsed value like this:
  match = Match(**record, metadata=parsed_metadata)
return matches
```

Reply:
I tried this code and ran into a problem regarding match having two metadata's. I found that deleting the metadata from the record after parsing worked along with appending the match to the matches list.